projects/vdk-core/tests/functional/run/test_run_sql_queries.py

```diff
@@ -241,3 +244,55 @@ def test_run_job_with_properties_and_sql_substitution_priority_order():
     assert db_plugin.db.execute_query("select * from test_table_override") == [
         ("one", 123)
     ]
+
+
+class DbOperationTrackPlugin:
+    def __init__(self):
+        self.log = []
+
+    @hookimpl(hookwrapper=True)
+    def db_connection_execute_operation(
+        self, execution_cursor: ExecutionCursor
+    ) -> Optional[int]:
+        self.log.append("start")
+        out: pluggy.callers._Result
+        out = yield
+        self.log.append(("end", out.excinfo is None))
```

Comment:
Because with two hooks you have less control and it's harder to use. 
We'd like to be able to write a plugin that tracks the duration of a query (for example(- the easiest way to do this is to use hookwrapper feature of pluggy (https://pluggy.readthedocs.io/en/stable/#wrappers)

This is not a new pattern. It is used in [run_job](https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/api/plugin/core_hook_spec.py#L145) and [run_step](https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/api/plugin/core_hook_spec.py#L171) hooks. 

See similar example in pytest (which is inspiration for this): https://docs.pytest.org/en/6.2.x/_modules/_pytest/hookspec.html#pytest_cmdline_main